include/af/cuda.h

```diff
@@ -117,5 +122,93 @@ static inline void setNativeId(int nativeId)
 }
 #endif
 
+#if AF_API_VERSION >= 34
+/**
+   Interop function that converts a thrust::device_vector to an af::array
+
+   \param[in] tvec thrust::device_vector to be converted
+   \return af::array with converted contents of the thrust vector
+
+   \ingroup cuda_mat
+ */
+template<typename T>
+static af::array array(thrust::device_vector<T> &tvec)
+{
+    af::array output;
+    if(tvec.empty()) return output;
+    T* ptr = thrust::raw_pointer_cast(tvec.data());
+    af::array tmp = af::array(tvec.size(), ptr, afDevice);
+    output = tmp.copy();
+    tmp.lock();
```

Comment:
Don't perform a copy by default. Have a flag that is set to false and copy when the user requests it.